packages/react/src/index.js

```diff
@@ -51,7 +51,7 @@ function Microlink (props) {
     isLoadingUndefined ? true : loadingProp
   )
   const [cardData, setCardData] = useState({})
-  const apiUrl = useMemo(() => createApiUrl(props), [props])
+  const [apiUrl, apiUrlProps] = useMemo(() => getApiUrl(props), [props])
```

Comment:
`getApiUrl` is returning two tings:

- `apiUrl` (string): The API url for doing the GET request.
- `apiUrlProps` (object): some options to be pass to `fetch` (like `headers`, etc).

Since this is directly used by `useCallback` props, I'm not sure if the best thing here is to split them into two different thing (the current approach) or pass it as a whole thing, like:


```js
const apiUrlProps = useMemo(() => getApiUrl(props), [props])

// passing it to fetcher
fetchFromApi(props.url, ...apiUrlProps)

// later, on props over `useCallbacks`
[canFetchData, setData, ...apiUrlProps])
```

thoughts? 🤔